pallets/confidential-asset/src/lib.rs

```diff
@@ -17,6 +17,78 @@
 //!
 //! The Confidential Asset module is one place to create the MERCAT security tokens on the
 //! Polymesh blockchain.
+//!
+//! ## Overview
+//!
+//! The following documentation covers the functionalities need for confidential transfer of a token.
+//! Part of this functionality (creating a confidential account and minting confidential tokens) are
+//! handled in this pallet, while the confidential transfer is handled by the
+//! [settlement module](../pallet_settlement/index.html). These pallets call out to the
+//! [MERCAT library]https://github.com/PolymathNetwork/cryptography) which is an implementation of the
+//! [MERCAT whitepaper](https://info.polymath.network/hubfs/PDFs/Polymath-MERCAT-Whitepaper-Mediated-Encrypted-Reversible-SeCure-Asset-Transfers.pdf).
+//!
+//!
+//! ### Terminology
+//!
+//! - The parties:
+//!   - Sender: The party who is sending tokens out of her account. We refer to her as Alice in the
+//!     examples.
+//!   - Receiver: The party who is receiving some tokens to his account. We refer to him as Bob in
+//!     the examples.
+//!   - Mediator: Also known as the exchange, is the party that preforms compliance checks and
+//!     approves/rejects the transaction. We refer to him as Charlie in the examples.
+//!   - Issuer: The party that mints the tokens into her account.
+//!
+//! - Phases of a successful transfer:
+//!   - Mint/issue: deposit tokens to an account
+//!   - Initialize a transfer: Alice indicates that she want to transfer some tokens to Bob.
+//!   - Finalize a transfer: Bob acknowledges that he expects to receive tokens from Alice.
+//!   - Justify a transfer: Charlie approves the transfer.
+//!   - Verification: the chain verifies all the data and updates the encrypted balance of Alice and Bob.
+//!
+//! - The workflow: the overall workflow is that at different phases of the transaction, each party
+//!   performs a task locally (e.g., on their wallet), creates a cryptographic proof of their task,
+//!   and submits the proof to the chain for verification.
+//!
+//!   There are X different phases for a transaction
```

Comment:
We can improve this --- 

Procedure to generate a confidential-asset and p2p transfer process - 

1. `create_confidential_asset()` dispatchable allows to create a confidential asset but not allow minting atomically in the same transaction as `create_asset()` do for non-confidential asset creation.
2. Different parties can create confidential accounts to manage confidential asset in their wallet and submit
     the proof of correctness.
        - The chain verifies the proofs and stores the confidential account on the chain.
3. Issuer can issue tokens of a confidential asset using `mint_confidential_asset()` dispatch and submits the proof of correctness to the chain.
       - The chain verifies the proofs and updates the encrypted balance on the chain.
4. Sender initiates a transfer in its wallet and submits the proof of correctness to the chain.
5. Receiver finalizes a transfer in its wallet and submits the proof of correctness to the chain.
6. Mediator justifies a transfer in its wallet and submits the proof of correctness to the chain.
7. Once all proofs of steps 4-6 are gathered, the chain verifies them and updates the encrypted balance of the Sender and the Receiver.

NB - Steps 2 and 3 can be performed in any order.

Reply:
I used your wording but made some modifications to further clarify it.